### PR TITLE
Removing explain_prediction 

### LINKED ISSUE #1832: Remove explain_prediction()
Now that explain_prediction() has been deprecated and its functionality moved into explain_predictions(), the outcome of this issue is that explain_prediction() and all its references are removed from the baseline.

### PR DESCRIPTION
**Pull request by freddyaboulton**
### Pull Request Description
Fixes #1832

-----
*After creating the pull request: in order to pass the **release_notes_updated** check you will need to update the "Future Release" section of* `docs/source/release_notes.rst` *to include this pull request by adding :pr:`123`.*

### CODE REVIEW

File: evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py [hunk 1]
```diff
@@ -28,213 +26,6 @@ def compare_two_tables(table_1, table_2):
         assert row.strip().split() == row_answer.strip().split()
 
 
-test_features = [[1], np.ones((15, 1)), pd.DataFrame({"a": [1, 2, 3], "b": [1, 2, 3]}).iloc[0],
```
Comment (freddyaboulton) [inline evalml/tests/model_understanding_tests/prediction_explanations_tests/test_explainers.py:31]: We have coverage in `test_explain_predictions_best_worst_and_explain_predictions`

**Review by jeremyliweishih — APPROVED** [review]
nice LGTM

**Review by angela97lin — APPROVED** [review]
Noice, thanks for coming back to this!

**Review by bchen1116 — APPROVED** [review]
🚀

**Pull request merged**